lib/pry/pry_class.rb

```diff
@@ -5,14 +5,22 @@ class Pry
   HOME_RC_FILE =
     if ENV.key?('PRYRC')
       ENV['PRYRC']
-    elsif File.exist?(File.expand_path('~/.pryrc'))
-      '~/.pryrc'
-    elsif ENV.key?('XDG_CONFIG_HOME') && ENV['XDG_CONFIG_HOME'] != ''
-      # See XDG Base Directory Specification at
-      # https://standards.freedesktop.org/basedir-spec/basedir-spec-0.8.html
-      ENV['XDG_CONFIG_HOME'] + '/pry/pryrc'
     else
-      '~/.config/pry/pryrc'
+      pryrc = begin
+                File.expand_path('~/.pryrc')
+              rescue ArgumentError
```

Comment:
How do I manually trigger this? I tried:

```
[1] pry(main)> ENV['HOME']
=> "/Users/kyrylo"
[2] pry(main)> ENV['HOME'] = nil
=> nil
[3] pry(main)> File.expand_path('~/.pryrc')
=> "/Users/kyrylo/.pryrc"
[4] pry(main)> ENV['HOME']
=> nil
```

Seems to be working without raising. I also tried unsetting $HOME in the shell. Also works fine.

Reply:
```
> env -i ruby -e "File.expand_path('~/.pryrc')"
-e:1:in `expand_path': couldn't find HOME environment -- expanding `~' (ArgumentError)
	from -e:1:in `<main>'
```
reproduces it for me.

Reply:
Lol, shit gets complicated. It's got like numerous backup ways to try and figure out your home directory, I think cheister's works b/c it blows away the `PATH` var which causes it to find the system Ruby which, for reasons I don't understand, can't figure it out! (This Ruby Tapas was really helpful as I was scratching my head about this). https://www.rubytapas.com/2012/10/15/episode-010-finding-home/